src/plugins/inputs/msg.js

```diff
@@ -1,9 +1,70 @@
 "use strict";
 
-exports.commands = ["msg", "say"];
+const Chan = require("../../models/chan");
+const Msg = require("../../models/msg");
+
+exports.commands = ["query", "msg", "say"];
+
+function parseInput(cmd, args, chan) {
+	switch (cmd) {
+	case "msg":
+	case "query":
+		return args.shift();
+	default:
+		return chan.name;
+	}
+}
 
 exports.input = function(network, chan, cmd, args) {
-	const target = cmd === "msg" ? args.shift() : chan.name;
+	const target = parseInput(cmd, args, chan);
```

Comment:
Pretty picky point here, but maybe this should be `getTarget` rather than `parseInput`, just because it's not parsing all the input, it's just getting the target.